### PR TITLE
chore: upgrades minor and patch deps

### PR DESCRIPTION
**Pull request by madou**
This actually shows the type errors we saw when doing your interpolation changes @pgmanutd.

This is a pre-req for the `runtime: automatic` jsx fix.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (pgmanutd) [inline .vscode/settings.json:2]: Nice, we configured it here. I had similar one in my user settings.

File: package.json [hunk 2]
```diff
@@ -31,14 +31,14 @@
     "version-nightly": "bash scripts/should-nightly-run.sh && npm version prerelease --no-git-tag-version --preid=$(git rev-parse HEAD)",
     "postversion": "bash scripts/push-if-git.sh",
     "bundlesize": "yarn build && yarn build:dead-code-elimination && size-limit",
-    "postinstall": "npx yarn-deduplicate"
+    "postinstall": "npx yarn-deduplicate && yarn --ignore-scripts"
```
Comment (pgmanutd) [inline package.json:34]: Are we ignoring postinstall script for dependencies? Any specific reason? 🤔

File: packages/babel-plugin/src/utils/ast-builders.tsx [hunk 3]
```diff
@@ -46,12 +46,16 @@ const hoistSheet = (sheet: string, meta: Metadata): t.Identifier => {
...
 
   const sheetIdentifier = meta.parentPath.scope.generateUidIdentifier('');
-  const parent = meta.parentPath.findParent((path) => path.isProgram()).get('body') as NodePath[];
-  const path = parent.filter((path) => !path.isImportDeclaration())[0];
-
-  path.insertBefore(
-    t.variableDeclaration('const', [t.variableDeclarator(sheetIdentifier, t.stringLiteral(sheet))])
-  );
+  const parent = meta.parentPath.findParent((path) => path.isProgram());
+  const parentBody = parent && (parent.get('body') as NodePath[]);
```
Comment (pgmanutd) [inline packages/babel-plugin/src/utils/ast-builders.tsx:50]: So all of these started breaking just after patch/minor bump ups? 😨

File: packages/react/src/codemods/codemods-helpers.tsx [hunk 4]
```diff
@@ -21,11 +23,14 @@ export const getImportDeclarationCollection = ({
   j: JSCodeshift;
   collection: Collection<any>;
   importPath: string;
-}) =>
-  collection
+}): Collection<ImportDeclaration> => {
```
Comment (pgmanutd) [inline packages/react/src/codemods/codemods-helpers.tsx:26]: Love explicit typings over implicit one. TS compiler will be fast if we specify return types.

File: packages/babel-plugin/src/utils/ast-builders.tsx (hunk 3 above)
Comment (madou) [reply packages/babel-plugin/src/utils/ast-builders.tsx:50]: yeah

File: package.json (hunk 2 above)
Comment (madou) [reply package.json:34]: only for postinstall - this is to stabalize the yarn lock after de-duping (and so it doesnt result in an infinite loop)

unfortunately there can be instances where running yarn after yarn de-dupe will result in the yarn lock file updating again

File: packages/react/src/codemods/codemods-helpers.tsx (hunk 4 above)
Comment (madou) [reply packages/react/src/codemods/codemods-helpers.tsx:26]: we could add an eslint rule for that too.

Comment (pgmanutd) [reply packages/react/src/codemods/codemods-helpers.tsx:26]: Maybe later. Should we add card for this? Because it might require changes too.

Comment (madou) [reply packages/react/src/codemods/codemods-helpers.tsx:26]: added https://github.com/atlassian-labs/compiled/issues/422

**Pull request merged**